### DISCUSSION TITLE
Export regions

### DISCUSSION THREAD

**Discussion by mrtripie** [Orama-Interactive/Pixelorama#812] [Ideas]
An export region would allow you to export only a rectangular region of the whole canvas. This would be represented by a box that is drawn over the canvas similar to guides. This could be useful as an alternative to #779 to allow the user to have part of their canvas outside of what would be exported.

- A possible option would be to allow whatever is not inside an export region to be covered by a semi-transparent overlay (wouldn't be needed for a first version though)

![image](https://user-images.githubusercontent.com/65431647/212488019-510017de-830e-4256-a6ee-b178cc30a7ea.png)

- Another possibility would be to have multiple export regions, with names assigned to each one. This would allow for both to be exported at once, with the region's name appended/prepended to the export names.
  - For example exporting art.png with two export regions may actually export:
    - art_region1.png
    - art_region2.png
  - Region names could be another optional overlay

![image](https://user-images.githubusercontent.com/65431647/212488711-340bfea5-9a04-46be-bbd4-9e1e3e54f070.png)

- Adding/editing the regions could work one of 2 ways:
  - A dedicated toolbar tool
  - A panel similar to the one for reference images

**Comment by dphaldes** [comment]
small addition: along with export the selected area, I would also like if I could extract selection into a new tab/project
